### PR TITLE
Retry MGS metrics checks if timeseries are missing

### LINKED ISSUE #6454: test failed in CI: omicron-nexus::test_all integration_tests::metrics::test_mgs_metrics
This test failed on a CI run on **"main"**:

https://github.com/oxidecomputer/omicron/runs/29322308970

Log showing the specific test failure:

https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC

Link here to the specific line of output from the buildomat log showing the failure:

https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5445

Excerpt from the log showing the failure:

```
[5472](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5472)	2024-08-27T17:26:33.010Z	=== checking timeseries for hardware_component:temperature ===
[5473](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5473)	2024-08-27T17:26:33.010Z	
[5474](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5474)	2024-08-27T17:26:33.010Z	thread 'integration_tests::metrics::test_mgs_metrics' panicked at nexus/tests/integration_tests/metrics.rs:316:13:
[5475](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5475)	2024-08-27T17:26:33.010Z	could not parse timeseries query response: parsing response body
[5476](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5476)	2024-08-27T17:26:33.010Z	
[5477](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5477)	2024-08-27T17:26:33.010Z	Caused by:
[5478](https://buildomat.eng.oxide.computer/wg/0/details/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC#S5478)	2024-08-27T17:26:33.010Z	    missing field `tables` at line 5 column 1
```

### PR DESCRIPTION
**Pull request by hawkw**
The test `integration_tests::metrics::test_mgs_metrics` has failed flakily a few times on `main` and PRs (see #6454).

This is due to a race condition where we attempt to test that the expected timeseries exist when they haven't yet been collected. The test waits for the Oximeter producer endpoint to be *registered*, and then immediately triggers the metrics collection: https://github.com/oxidecomputer/omicron/blob/a0cdce7395cec5813fc697172b8a28a69d27093d/nexus/tests/integration_tests/metrics.rs#L741-L745

However, if we look at the logs, we see that when we trigger Oximeter to collect the metrics, the SP pollers haven't produced any samples yet. We can see that Oximeter does collect metrics from the MGS instance, it just doesn't have anything yet:

https://buildomat.eng.oxide.computer/wg/0/artefact/01J6AC5R3TN7HA7J5XR46XM9JM/nxMhUQ2DddPGyS7EJUzyESwgw1GGhoLMU8bPmZEIGVtDZ9q6/01J6AC6HPHGKFQN3CY1YJ15JEC/01J6AF1QQQNVH0P76YGCVTX5JX/test_all-33bc5def8da77203-test_mgs_metrics.130753.4.log?format=x-bunyan#L106

This commit fixes this issue by wrapping the code that checks whether the expected timeseries exist in a `poll::wait_for_condition`, allowing them to be retried if some (or all) timeseries are not yet present. We force Oximeter to collect any new samples on each retry. If we encounter an unexpected timeseries, the test panics immediately, but if some expected timeseries aren't present, we retry for up to 30 seconds. If they're still not there, *then* the test fails.

Fixes #6454

### CODE REVIEW

File: nexus/tests/integration_tests/metrics.rs [hunk 1]
```diff
@@ -690,60 +690,101 @@ async fn test_mgs_metrics(
...
+            // We really don't need to wait that long to know that the sensor
+            // metrics will never be present. This could probably be shorter
+            // than 30 seconds, but I want to be fairly generous to make sure
+            // there are no flaky failures even when things take way longer than
+            // expected...
+            &Duration::from_secs(30),
+        )
+        .await;
+        if result.is_err() {
+            panic!("failed to find timeseries for {query} within 60s")
```
Comment (bnaecker) [inline nexus/tests/integration_tests/metrics.rs:719]: Might be more helpful if the message said something like, "when running query '{query}' within 30s"? It _is_ 30s, right?

**Review by bnaecker — APPROVED** [review]
Looks good! One small suggestion on an error string, but take it or leave it. Thanks!

File: nexus/tests/integration_tests/metrics.rs (hunk 1 above)
Comment (hawkw) [reply nexus/tests/integration_tests/metrics.rs:719]: whoops, good catch! thank you!

Comment (hawkw) [reply nexus/tests/integration_tests/metrics.rs]: fixed in d2df64160c9559fa99336ad40d793c38db4ecc4a :)

**Pull request merged**